homeassistant/components/vizio/media_player.py

```diff
@@ -37,32 +40,47 @@
 MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
 
-SUPPORTED_COMMANDS = SUPPORT_TURN_ON | SUPPORT_TURN_OFF \
-                     | SUPPORT_SELECT_SOURCE \
-                     | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK \
-                     | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP \
-                     | SUPPORT_VOLUME_SET
+SUPPORTED_COMMANDS = {
+    'soundbar': (SUPPORT_PLAY | SUPPORT_PAUSE
+                 | SUPPORT_SELECT_SOURCE
+                 | SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+                 | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP
+                 | SUPPORT_VOLUME_SET),
+    'tv': (SUPPORT_PLAY | SUPPORT_PAUSE
+           | SUPPORT_SELECT_SOURCE
+           | SUPPORT_TURN_ON | SUPPORT_TURN_OFF
+           | SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK
+           | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP
+           | SUPPORT_VOLUME_SET)
+}
+
+MAX_VOLUME = {
+    'soundbar': 31.0,
+    'tv': 100.0
+}
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_HOST): cv.string,
-    vol.Required(CONF_ACCESS_TOKEN): cv.string,
+    vol.Optional(CONF_ACCESS_TOKEN): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_SUPPRESS_WARNING, default=False): cv.boolean,
+    vol.Optional(CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS):
+        vol.All(cv.string, vol.Lower, vol.In(['tv', 'soundbar'])),
```

Comment:
Would it be possible to detect the device type from device responses, avoiding the need to configure it manually?

Reply:
I don't have enough data to do this deterministically. I can guess at how to do that but this is IMO the safest approach.

Reply:
Ok, fine for me.